packages/mui-utils/src/getReactElementRef/getReactElementRef.test.tsx

```diff
@@ -0,0 +1,39 @@
+import { expect } from 'chai';
+import getReactElementRef from '@mui/utils/getReactElementRef';
+import * as React from 'react';
+
+describe('getReactElementRef', () => {
+  it('should return undefined when not used correctly', () => {
```

Comment:
I wonder if it'd be better to always return `null` in these cases.

Reply:
No strong opinion, logic might need to change from `element?.ref` to `element?.ref??null` and `element.props?.ref` to `element.props?.ref??null`. If this change doesn't alter bundle size, i'm okay to do the change

Reply:
Nah, keep as is. We can always come back to it if needed.